### PR TITLE
Add configuration flow to Google Calendar

### PR DESCRIPTION
**Pull request by allenporter**
## Proposed change

Add instructions with details about the new configuration flow, as well as screenshots and better documentation for the existing OAuth flow.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [X] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: https://github.com/home-assistant/core/pull/68010
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [X] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [ ] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

**Review by MartinHjelmare — DISMISSED** [review]

**Comment by allenporter** [comment]
Next steps from discussion in https://github.com/home-assistant/home-assistant.io/pull/22027 is that @frenck will merge `source/_integrations/google.markdown` changes from `current` into `next. Then I will rebase this.

**Comment by frenck** [comment]
@allenporter `current` -> `next` ✅

**Comment by allenporter** [comment]
Thank you, that was very helpful, this has now been rebased.

File: source/_integrations/google.markdown [hunk 1]
```diff
@@ -78,16 +78,60 @@ calendar_access:
...
+
+    ![Screenshot of Success](/images/integrations/google/success.png)
+
+{% enddetails %}
+
+## Troubleshooting
 
-In case you get an `Authentication code expired, please restart Home-Assistant and try again` error message, switch your timezone to `Etc/GMT` and restart Home Assistant. This should fix the issue and the `google_calendars.yaml` configuration file will be created.
-You can then switch back the timezone to your original one and restart Home Assistant again.
+If the set up process fails and you see an error message such as *Authentication code expired, please try again* you may want to try the flow again. You may also check the logs for additional error messages that may indicate a misconfiguration such as an invalid client id or secret.
```
Comment (frenck) [inline source/_integrations/google.markdown]:
```suggestion
If the setup process fails and you see an error message such as *Authentication code expired, please try again* you may want to try the flow again. You may also check the logs for additional error messages that may indicate a misconfiguration such as an invalid client id or secret.
```

File: source/_integrations/google.markdown [hunk 2]
```diff
@@ -78,16 +78,60 @@ calendar_access:
...
+{% details "OAuth and Device Authorization steps" %}
+
+1. The first step shows a link and a code.
+
+    ![Screenshot of Link Account](/images/integrations/google/link_account.png)
+
+1. Click on the link [https://www.google.com/device](https://www.google.com/device) to open a Google website which should open a new window where you can enter the code.
 
-If you are trying to switch to a new Google account then you would run into the following error message. Make sure to delete the existing **.google.token** file from your `config` folder and restart Home Assistant to try again.
+1. Home Assistant will wait for a short time while you complete the authorization steps, checking in the background for the authorization to be copmlete.
```
Comment (frenck) [inline source/_integrations/google.markdown]:
```suggestion
1. Home Assistant will wait for a short time while you complete the authorization steps, checking in the background for the authorization to be completed.
```

**Review by frenck — APPROVED** [review]
Thanks, @allenporter 👍

**Pull request merged**